pkg/controller/nodelifecycle/scheduler/taint_manager_test.go

```diff
@@ -206,18 +206,19 @@ func TestCreatePod(t *testing.T) {
 }
 
 func TestDeletePod(t *testing.T) {
-	stopCh := make(chan struct{})
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
 	fakeClientset := fake.NewSimpleClientset()
-	controller, _, _ := setupNewNoExecuteTaintManager(context.TODO(), fakeClientset)
+	controller, _, _ := setupNewNoExecuteTaintManager(ctx, fakeClientset)
 	controller.recorder = testutil.NewFakeRecorder()
-	go controller.Run(context.TODO())
+	go controller.Run(ctx)
```

Comment:
Case1: Use the contexture ctx rather than the empty one to prevent potential goroutine leakage.